packages/node_modules/@ciscospark/internal-plugin-ediscovery/src/ediscovery.js

```diff
@@ -0,0 +1,520 @@
+          if (res.body.length < size) {
+            return contentSummary;
+          }
+          offset += size;
+          return getNextContentSummaryPage(plugin);
+        })
+        .catch((reason) => {
+          plugin.logger.error(reason);
+          return reason;
+        });
+    }
+
+    return getNextContentSummaryPage(this);
+  },
+
+  /**
+   * Retrieves a list of the conversations relevant to a specified report, paged with a specified offset and size
+   *
+   * @param {UUID} reportId - Id of the report being requested
+   * @param {long} offset - the starting index of the results the caller wants
+   * @param {long} size - the number of conversations the caller wants
+   * @returns {Promise<ResponseEntity<ContentSummary>>} Http response containing the specified content summary
+   */
+  @waitForValue('@')
+  getPagedContentSummary(reportId, offset, size) {
+    if (!reportId) {
+      throw Error('Invalid parameter(s)');
+    }
+
+    return this.request({
+      method: 'GET',
+      service: 'ediscovery',
+      resource: `reports/${reportId}/contents/summary`,
+      qs: {offset, size}
+    })
+      .then((res) => res)
+      .catch((reason) => {
+        this.logger.error(reason);
+        return reason;
+      });
+  },
+
+  /**
+   * Generates the report content and triggers the download of the content
+   *
+   * @param {UUID} reportId - Id of the report being requested
+   * @returns {List<Activities>} HttpResponse
+   */
+  @waitForValue('@')
+  generateReport(reportId) {
+    if (!reportId) {
+      throw Error('Invalid parameter(s)');
+    }
+
+    let offset = 0;
+    const size = 100;
+    const reportGenerator = new ReportGenerator({reportId});
+
+    function decryptNextActivities(plugin) {
+      return plugin.request({
+        method: 'GET',
+        service: 'ediscovery',
+        resource: `reports/${reportId}/contents`,
+        qs: {offset, size}
+      })
+        .then((res) => {
+          const contents = res.body;
+          return plugin._decryptActivities(contents);
+        })
+        .then((activities) => {
+          if (activities && Array.isArray(activities) && activities.length > 0) {
+            reportGenerator.add(activities);
+            if (activities.length < size) {
+              // No more activities to decrypt
+            }
+            else {
+              // Add activities to the report, increment and decrypt next activities
+              offset += size;
+              decryptNextActivities(plugin);
+            }
+          }
+        })
+        .catch((reason) => {
+          plugin.logger.error(reason);
+          return reason;
+        });
+    }
+
+    return decryptNextActivities(this).then(() => reportGenerator.saveAs());
+  },
+
+  /**
+   * Decrypt the activities found in the report content
+   *
+   * @param {Object[]} activities An array of activities with encrypted messages
+   * @returns {Promise<[any]>} The activities with their messages decrypted
+   */
```

Comment:
Depending on the activity type, the content will de decrypted and in the case of shares and boards, downloaded.